src/styles/main.scss

```diff
@@ -0,0 +1,89 @@
+$border-style: 1px solid black;
+$borde-size: 1px;
+$box-size: 100px;
+$calendar-size: calc(($box-size + 1px) * 7 - 1px);
+$margin-left: calc($calendar-size - ($box-size + 1px));
+$standard-color: #eee;
+$chose-color: #ffbfcb;
+$gap: 1px;
+$week: (
+  mon: 0,
+  tue: calc($box-size + $gap),
+  wed: calc($box-size + $gap) * 2,
+  thu: calc($box-size + $gap) * 3,
+  fri: calc($box-size + $gap) * 4,
+  sat: calc($box-size + $gap) * 5,
+  sun: calc($box-size + $gap) * 6,
+);
+
+@keyframes example {
+  0% {
+    background-color: $standard-color;
+  }
+  100% {
+    background-color: $chose-color;
+  }
+}
+
+body {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  margin: 0;
+  font-size: 30px;
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+.calendar {
+  width: $calendar-size;
+  padding: 10px;
+  display: flex;
+  flex-flow: row wrap;
+  gap: $gap;
+
+  &__day {
+    display: flex;
+    box-sizing: border-box;
+    height: $box-size;
+    width: $box-size;
+    border: $border-style;
+    background-color: $standard-color;
+    justify-content: center;
+    align-items: center;
+    transition-duration: 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      animation-name: example;
+      transform: translateY(-20px);
+      animation-duration: 0.5s;
+      background-color: $chose-color;
+    }
+  }
+
+  @for $index from 1 through 31 {
+    &__day:nth-child(#{$index})::before {
+      content: '#{$index}';
+    }
+  }
+}
+
+@each $day, $margin in $week {
+  .calendar--start-day-#{$day} {
+    .calendar__day:first-child {
+      margin-left: $margin;
+    }
+  }
+}
+
+@for $days from 28 through 31 {
+  .calendar--month-length-#{$days} {
+    @for $i from ($days + 1) through 32 {
+      .calendar__day:nth-child(#{$i}) {
+        display: none;
+      }
+    }
+  }
+}
```

Comment:
you can achieve this result with only one for loop. [see how works :nth-child(an +b)   ](https://css-tricks.com/useful-nth-child-recipies/#aa-select-all-but-the-first-five)